ddl/partition.go

```diff
@@ -2448,6 +2455,14 @@ func (w *worker) onExchangeTablePartition(d *ddlCtx, t *meta.Meta, job *model.Jo
 		}
 	}
 
+	if variable.EnableCheckConstraint.Load() {
```

Comment:
I think this should also be under the `withValidation` check, i.e. if `WITHOUT VALIDATION` is given, we should trust the user and not start a big scan to validate the CHECK, just like we would skip checking that all rows match the partition.

Reply:
Yes, It is a good suggestion.  
But there is also a problem,  when withValidation is on, duraing write only state, there seems to no good method to check withValidation is on.   
And current Validation check also has the same problem.

Reply:
The `WITH[OUT] VALIDATION` is only for existing rows in the non-partitioned table (and with CHECK also for the partition). It does not need to control the insert/update that may concurrently happen, it is OK if those concurrent checks is always on. Ref [Exchange partition in MySQL Manual](https://dev.mysql.com/doc/refman/8.0/en/partitioning-management-exchange.html)